p4runtime_sh/shell.py

```diff
@@ -357,13 +357,15 @@ def __setitem__(self, name, value):
         fullname = self._full_field_name(name)
         field_info = self._get_mf(fullname)
         self._mk[fullname] = self._parse_mf(value, field_info)
-        print(self._mk[fullname])
+        if context.do_printing:
+            print(self._mk[fullname])
```

Comment:
I think it's typically better to have a custom print function that takes care of the check for you:
```python
def _print(*args, **kwargs):
    if xyz:
        print(*args, **kwargs)
```